libstorage/drivers/storage/ebs/utils/utils_unix.go

```diff
@@ -49,10 +51,27 @@ var (
 		},
 		DeviceRE: regexp.MustCompile(`^xvd[f-p]$`),
 	}
+	nvmeDeviceRange = &DeviceRange{
+		ParentLetters: []string{"0"},
+		ChildLetters: []string{
+			"0", "1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13",
+			"14", "15", "16", "17", "18", "19", "20", "21", "22", "23", "24", "25", "26"},
+		NextDeviceInfo: &types.NextDeviceInfo{
+			Prefix:  "nvme",
+			Pattern: "[0-26]n.*",
+			Ignore:  false,
+		},
+		DeviceRE: regexp.MustCompile(`^nvme[0-26]n.*`),
+	}
 )
 
 // GetDeviceRange returns a specified DeviceRange object
-func GetDeviceRange(useLargeDeviceRange bool) *DeviceRange {
+func GetDeviceRange(useLargeDeviceRange bool, instanceType string) *DeviceRange {
+	log.Debug("InstanceType: ", instanceType)
+	if strings.HasPrefix(instanceType, "c5") || strings.HasPrefix(instanceType, "m5") {
```

Comment:
i wonder if we could make this less static - i imagine m5 and other instances doing NVMe too going forward 

would it be sufficient to stat if `/dev/nvme0` exist ?

Reply:
I know i3's use nvme ephemeral drives, but I'm not certain how EBS might be presented to this family.  I'll investigate and follow up.

Reply:
this is from an `i3.2xlarge`

```
$ df -h
Filesystem      Size  Used Avail Use% Mounted on
udev             30G   12K   30G   1% /dev
tmpfs           6.0G   75M  6.0G   2% /run
/dev/xvda1       99G   15G   80G  16% /
none            4.0K     0  4.0K   0% /sys/fs/cgroup
none            5.0M     0  5.0M   0% /run/lock
none             30G     0   30G   0% /run/shm
none            100M     0  100M   0% /run/user
/dev/nvme0n1    1.8T  1.6T  149G  92% /data
```

`/data` is the local instance data
`/dev/xvda1` is the EBS root volume

Reply:
my main concern is that having rexray hardcode instance families for using nvme or not is kinda rough adoption wise, since rexray don't cut releases that often

Reply:
@jippi apologies that it has been so long, indeed. But we are ramping back up. =)

I need to come up to speed on how NVMe devices are working on AWS before providing a proper review. It does appear complicated if they are handled differently depending on the instance type...  :/

Reply:
with the `udev` rule in OP, that is also bundled in AWS Linux handles mapping the nvme EBS device to the classic `/dev/*vda*` device - so if reading the type of the rexray expected `/dev/` device and checking if its a symlink to a nvme device - it should be generic enough to handle all instance types.

The reason why it does not work atm is that `/proc/partitions` show the nvme device, not the `virtual` device the `udev` rule creates - if its modified to `resolve` the non-nvme device name to the real nvme device name, it should just work (tm)